### ISSUE TITLE
Doesn't work at all with recent pino versions

### ISSUE THREAD

**Issue by sammko** [Julien-R44/pino-loki#14]
Let me know if I'm doing something wrong, but the following program

```js
import { pino } from "pino";

const transport = pino.transport({
  target: 'pino-loki',
  options: { host: 'http://localhost:3100/', labels: { app: 'test' } }
})
const log = pino(transport)

for (let i = 0; i < 10; i++) {
  log.info({ i }, "asdf")
}

log.flush()
```

does not send anything to Loki. Other transports such as `pino-pretty` or `pino/file` work correctly. This is in a clean environment. I've tried an assortment of node versions from 17 to 20.

Downgrading to pino 8.0.0 (almost a year old) and setting "batching: false" in the options seems to help. I found this to be the breaking commit in pino, between 8.0 and 8.1. https://github.com/pinojs/pino/commit/0395233031d8d23b438ed5a72e4bc411b4261d7c

**Comment by sammko** [comment]
Ah, adding a

```js
setTimeout(() => { }, 2000)
```

at the end and "batching: false" also works with recent pino versions and so does e.g. "batching: true, interval: 1".

So maybe this is the same issue as #7 after all.

**Comment by Julien-R44** [comment]
Gonna take a look asap. thanks !

**Comment by Julien-R44** [comment]
Should be fixed with 2.1.0. There was indeed an issue with it. Worker thread executing pino-loki was killed without sending the pending batch

Feel free to re-open if issue persist for you
